### DISCUSSION TITLE
Synology SSO Server for Wiki.js

### DISCUSSION THREAD

**Discussion by Frixs** [requarks/wiki#6955] [Help / Questions]
**Hello,** 
**I would like to make available the SSO option to my Synology NAS.**

I run Portainer on my Synology NAS in which my WIki.js lives.
On the same Synology NAS, I would like to make an SSO Server for my WIki.js, so Synology users can simply log into my WIki.js.
I run everything in my local network only.

On the Synology side, I use OIDC like so:
<img width="640" alt="image" src="https://github.com/requarks/wiki/assets/17912365/cb7bebe7-09de-443e-a404-9fb930379b6f">
<img width="667" alt="image" src="https://github.com/requarks/wiki/assets/17912365/c989fe86-1b40-48a1-b4ea-76f9ad4d553e">
For the redirect URI I use the one provided on the WIki.js side from the Configuration Reference tab.
<img width="313" alt="image" src="https://github.com/requarks/wiki/assets/17912365/27cece3a-b3cd-451c-880e-fe7bac735f61">
<img width="681" alt="image" src="https://github.com/requarks/wiki/assets/17912365/5b37a833-e0be-49e1-a839-a4c8152737fb">
This is Well-Known URL content provided:
<img width="672" alt="image" src="https://github.com/requarks/wiki/assets/17912365/bee740ac-53b5-40fc-9a47-5e43c918d21c">

This is my SSO setting on the Wiki.js side:
<img width="644" alt="image" src="https://github.com/requarks/wiki/assets/17912365/2ffa94d1-dd78-4cc6-b7be-ad08514e0370">

It seems to work on the Synology side:
<img width="859" alt="image" src="https://github.com/requarks/wiki/assets/17912365/bacabe41-e799-4ae4-ab2c-4a2870e87ce6">

However, on the Wiki.js, I cannot sign in properly:
On the first login try, I always get this:
<img width="203" alt="image" src="https://github.com/requarks/wiki/assets/17912365/2dbddd6d-fc37-4de2-bd4c-616a54c3c8ce">
When I am already logged in to the Synology NAS account in my browser (press F5), this is the issue:
<img width="207" alt="image" src="https://github.com/requarks/wiki/assets/17912365/9cd4a63e-7f90-4074-a49d-ebb4aa81ced9">

**Has anyone tried to connect Wiki.js with Synology NAS? Any help? I appreciate any kind of guidance/help.**
The WIki.js uses email to log in while Synology uses a username., not sure if this can be an issue. I guess not, but I am starting to be helpless.

The "testuser" is pre-created by me in the WIki.js with the same email (as he has in Synology) and set to Generic OIDC / OAuth2 provider.

**Comment by Frixs** [comment]
**SOLVED**
If anyone runs into this, I found a fix. SSL was the trouble maker here.
It is important to keep either HTTP or HTTPS, but do not mix it (no matter what it suggests to you).

<img width="403" alt="image" src="https://github.com/requarks/wiki/assets/17912365/2e8ca672-6b65-49e2-928d-c7159de3c8c4">

![image](https://github.com/requarks/wiki/assets/17912365/71115de1-be92-4694-91d5-9c528400248c)

![image](https://github.com/requarks/wiki/assets/17912365/f43c5e00-52c2-4293-9997-8ce96ceccbbe)

**Comment by SmolSoftBoi** [comment]
Some of these settings appear to have changed in the last year, could you share your working settings please? I'm trying to accomplish the same thing.

**Reply by Frixs** [reply]
Hello, what are you currently struggling with? I did not touch it from the time I started to use it. Let me know, I can try to assist you. :-)

Anyway at this time I'm moving from WIki.js to Outline. It's a much more convenient and modern wiki editor compared to this one.
